src/canisters/telemetry/src/telemetry_backend/main.mo

```diff
@@ -0,0 +1,208 @@
+import Debug "mo:base/Debug";
+import Hash "mo:base/Hash";
+import Int "mo:base/Int";
+import Nat "mo:base/Nat";
+import Option "mo:base/Option";
+import Prelude "mo:base/Prelude";
+import Time "mo:base/Time";
+import Trie "mo:base/Trie";
+
+import Aggregation "Aggregation";
+import CommandDetails "CommandDetails";
+import CommandResults "CommandResults";
+import Data "Data";
+import InvocationDetails "InvocationDetails";
+import Interface "Interface";
+
+actor {
+  type Time = Time.Time;
+  type Trie<K, V> = Trie.Trie<K, V>;
+  type Trie2D<K1, K2, V> = Trie.Trie2D<K1, K2, V>;
+
+  type AggregationPeriodStart = Aggregation.AggregationPeriodStart;
+  type CommandDetails = CommandDetails.CommandDetails;
+  type InvocationDetails = InvocationDetails.InvocationDetails;
+  type CommandResultsEntry = Interface.CommandResultsEntry;
+  type CommandResultCounts = CommandResults.CommandResultCounts;
+  type CommandSuccessRatesEntry = Interface.CommandSuccessRatesEntry;
+  type ReportCommandArgs = Interface.ReportCommandArgs;
+  type V0 = Data.V0;
+
+  let nsPerDay = 86_400 * 1000_000_000;
+  let nsPer30Days = 30 * nsPerDay;
+
+  stable var v0 : ?V0 = ?Data.v0();
+  let data : V0 = switch (v0) {
+    case null Prelude.unreachable();
+    case (?v0) v0;
+  };
+
```

Comment:
Reviewers: See https://github.com/dfinity/sdk/commit/eba34882a276dc48225b3156be20eb3c3aeeef30#diff-c95f31a8341300c4f8213b1fb1d534d40d4dfee78d7bf9d4ff2ef6796bd3d699R46 for for upgrades will work